scripts/apply-i18n-batch.py

```
@@ -0,0 +1,57 @@
+        for key, translation in entries.items():
+            pattern = re.compile(r'("' + re.escape(key) + r'": ")((?:[^"\\]|\\.)*)(")')
+            found = pattern.findall(text)
+            if len(found) != 1:
+                sys.exit(f"{locale}: key {key!r} found {len(found)} times, expected 1")
+            current = found[0][1]
+            if "NEEDS TRANSLATION" not in current and not force:
+                sys.exit(
+                    f"{locale}: key {key!r} is already translated; refusing to overwrite "
+                    f"(pass --force if that is really what you want)"
+                )
+            text = pattern.sub(lambda m: m.group(1) + esc(translation) + m.group(3), text, count=1)
```

Ruff (0.16.0)

[warning] Function definition does not bind loop variable `translation`

(B023)